testnet/stacks-node/src/nakamoto_node/relayer.rs

```diff
@@ -0,0 +1,907 @@
+            local_peer,
+
+            last_network_block_height: 0,
+            last_network_block_height_ts: 0,
+            last_network_download_passes: 0,
+            min_network_download_passes: 0,
+            last_network_inv_passes: 0,
+            min_network_inv_passes: 0,
+
+            relayer,
+
+            miner_thread: None,
+            is_miner,
+            next_initiative: Instant::now() + Duration::from_secs(10),
+            last_committed: None,
+        }
+    }
+
+    /// have we waited for the right conditions under which to start mining a block off of our
+    /// chain tip?
+    fn has_waited_for_latest_blocks(&self) -> bool {
+        // a network download pass took place
+        (self.min_network_download_passes <= self.last_network_download_passes
+        // a network inv pass took place
+        && self.min_network_download_passes <= self.last_network_download_passes)
+        // we waited long enough for a download pass, but timed out waiting
+        || self.last_network_block_height_ts + (self.config.node.wait_time_for_blocks as u128) < get_epoch_time_ms()
+        // we're not supposed to wait at all
+        || !self.config.miner.wait_for_block_download
+    }
+
+    /// Handle a NetworkResult from the p2p/http state machine.  Usually this is the act of
+    /// * preprocessing and storing new blocks and microblocks
+    /// * relaying blocks, microblocks, and transacctions
+    /// * updating unconfirmed state views
+    pub fn process_network_result(&mut self, mut net_result: NetworkResult) {
+        debug!(
+            "Relayer: Handle network result (from {})",
+            net_result.burn_height
+        );
+
+        if self.last_network_block_height != net_result.burn_height {
+            // burnchain advanced; disable mining until we also do a download pass.
+            self.last_network_block_height = net_result.burn_height;
+            self.min_network_download_passes = net_result.num_download_passes + 1;
+            self.min_network_inv_passes = net_result.num_inv_sync_passes + 1;
+            self.last_network_block_height_ts = get_epoch_time_ms();
+            debug!(
+                "Relayer: block mining until the next download pass {}",
+                self.min_network_download_passes
+            );
+            signal_mining_blocked(self.globals.get_miner_status());
+        }
+
+        let net_receipts = self
+            .relayer
+            .process_network_result(
+                &self.local_peer,
+                &mut net_result,
+                &mut self.sortdb,
+                &mut self.chainstate,
+                &mut self.mempool,
+                self.globals.sync_comms.get_ibd(),
+                Some(&self.globals.coord_comms),
+                Some(&self.event_dispatcher),
+            )
+            .expect("BUG: failure processing network results");
+
+        if net_receipts.num_new_blocks > 0 || net_receipts.num_new_confirmed_microblocks > 0 {
```

Comment:
Does this also check Nakamoto blocks?

Reply:
This method is unchanged from the neon_node's `process_net_result` method.